apps/admin/backend/src/app.ts

```diff
@@ -198,12 +198,26 @@ function buildApi({
     },
     /* c8 ignore stop */
 
+    async getUsbDriveStatus(): Promise<UsbDriveStatus> {
+      return usbDrive.status();
+    },
+
+    ejectUsbDrive(): Promise<void> {
+      return usbDrive.eject();
+    },
+
+    async formatUsbDrive(): Promise<Result<void, Error>> {
+      try {
+        await usbDrive.format();
+        return ok();
+      } catch (error) {
+        return err(error as Error);
+      }
```

Comment:
I made this into a `Result` endpoint because, even wrapped in a `try/catch` block on the frontend, errors from `mutateAsync` would go to the error boundary but skip the `catch` block. I didn't realize it worked like that. And I believe the recommendation is to use the `Result` type here anyway.

I could move the `Result` return value down a layer into `libs/usb-drive`, but since this is currently the only consumer it doesn't really matter.

Reply:
If we're handling these errors explicitly to show messages to users on the frontend, then I agree `Result` is the right move!

Reply:
Also if it's true that exceptions thrown from `mutateAsync` don't bubble up in the original call stack but only go to the error boundary, that would be great news for us, because we could stop using `mutate`. The original reason I recommended using `mutate` over `mutateAsync` was to prevent uncaught exceptions from crashing tests with no useful error output, which was nearly impossible to debug and would happen whenever a mock function assertion failed.

Would you be down to explore this a bit more?

Reply:
OK yeah, I'll give this a try later today or tomorrow and post on Slack what I find.